### ISSUE TITLE
Fix definitions and naming conventions of LOSS functions in code. 

### ISSUE THREAD

**Issue by napsternxg** [deeplearning4j/deeplearning4j#940]
I was going through the definitions of the various loss functions in https://github.com/deeplearning4j/deeplearning4j/blob/79b171956f1ce1496a62f72220bb5143b7101fe9/deeplearning4j-core/src/main/java/org/deeplearning4j/nn/layers/BaseOutputLayer.java
And I found that RMSE_XENT is just RMSE and there is no cross entropy part in it. I was thinking it is a sum of RMSE and XENT errors. Don't you think the notation should just be simple RMSE as it is not related to cross entropy loss.

Also NEGATIVELOGLIKELIHOOD is same as MCXENT

I think MCXENT will should be CROSS ENTROPY and XENT should be BINARY CROSS ENTROPY

Using the convention from libraries in other languages would be best. I writing the issue details right now. Will work on the PR.

**Comment by napsternxg** [comment]
This is a good place to start about the names of Loss functions we should include. 
https://en.wikipedia.org/wiki/Loss_functions_for_classification

https://github.com/JohnLangford/vowpal_wabbit/wiki/Loss-functions

https://github.com/torch/nn/blob/master/doc/criterion.md#nn.ClassNLLCriterion

http://keras.io/objectives/

Most of the popular DL libraries like nn in torch and keras in python have loss function implementation on their side whereas torch and theano have no loss function implementations.

**Comment by nyghtowl** [comment]
Thanks for submitting the issue. NEGATIVELOGLIKELIHOOD is a common term used in CNN research while CROSS ENTROPY is a common term used in RNN. Thus, we are keeping both to make it easier for users coming in for different nets.

Changing names works. Just be sure to deprecate the old enums and update the javadoc when the PR is developed. Coping @AlexDBlack for any additional input.

**Comment by napsternxg** [comment]
Even if we have to keep the different names then also there should be just one implementation of the code. So the switch case block should use the same code for all the loss functions which have different names (hence, ENUM values) but have the same implementation.

**Comment by nyghtowl** [comment]
Not disagreeing there. It was like that at one point but apparently got changed along the way.

**Comment by AlexDBlack** [comment]
yep, mcxent and negativeloglikelihood should have same implementation (two cases in switch with same code). Used to be the same as @nyghtowl said.

We can't just remove/rename the old enums (otherwise we break other peoples code/config). Instead: deprecate the old enums (and add a javadoc comment on them: why + what to use instead), then add the new ones.

**Comment by agibsonccc** [comment]
Fixed already
